src/components/SamlConfiguration/index.jsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Collapsible, Icon } from '@edx/paragon';
+import SamlProviderConfigForm from './SamlProviderConfigForm';
+import SamlProviderDataForm from './SamlProviderDataForm';
+import { camelCaseObject, snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import LoadingMessage from '../LoadingMessage';
+import ErrorPage from '../ErrorPage';
+
+class SamlConfiguration extends React.Component {
+  state = {
+    providerConfig: undefined,
+    providerData: undefined,
+    error: undefined,
+    loading: true,
+    deleteEnabled: false,
+  };
+
+  componentDidMount() {
+    LmsApiService.getProviderConfig(this.props.enterpriseId)
+      .then((response) => {
+        this.setState({
+          providerConfig: response.data.results[0],
+          loading: false,
+        });
+      })
+      .catch(error => this.setState({
+        error,
+      }));
+    LmsApiService.getProviderData(this.props.enterpriseId)
+      .then((response) => {
```

Comment:
Curious if you considered combining these calls with `Promise.all()` and have a single `setState` for `loading`, `providerConfig`, and `providerData`?

As is, I believe one call could resolve and set `loading: false` before the other has finished.

Reply:
Ah! I am not familiar with Promise.all(). I will look into that.

> As is, I believe one call could resolve and set loading: false before the other has finished.

I was unfortunately aware of this bit but wasn't sure how to handle it, so this seems like a good direction!